src/mentionPlugin/MentionSearch/index.js

```diff
@@ -14,12 +14,16 @@ export default (mentions) => {
     };
 
     // Get the first 5 mentions that match
-    getMentionsForFilter = () => (
-      mentions
+    getMentionsForFilter = () => {
 
-      // TODO better search algorithm than startsWith
-      // mentions.filter((mention) => mention.get('handle').startsWith(this.mentionSearch)).slice(0, 10)
-    );
+      const userEntered = this.props.editor.props.editorState.getCurrentContent().getLastBlock().getText();
```

Comment:
While the filtering works well it' pretty much broken since the mention doesn't need to be the last `@`.